src/helpers/OAuthClient.js

```diff
@@ -0,0 +1,54 @@
+const got = require('got');
+const Cache = require('ttl');
+const qs = require('querystring');
+
+const cache = new Cache({
+  ttl: 3600 * 1000
+});
+if (process.env.NODE_ENV !== 'production') {
+  cache.on('hit', (key, val) => {
+    console.log(`Cache hit for key: ${key}; Value: ${val}`);
+  });
+  cache.on('miss', (key) => {
+    console.log(`Cache miss for key: ${key}`);
+  });
+  cache.on('put', (key, val, ttl) => {
+    console.log(`Cache put for key: ${key}; Value: ${val} with ttl: ${ttl}`);
+  });
+}
+
+async function getAccessToken(authSettings) {
+  let token = null;
+  if (authSettings.oauthConfig) {
+    const { tokenUrl, clientId, clientSecret } = authSettings.oauthConfig;
+    token = cache.get(clientId);
+    if (!token) {
+      const payload = { grant_type: 'client_credentials', client_id: clientId };
+      const cred = Buffer.from(`${clientId}:${clientSecret}`, 'utf8').toString('base64');
+      console.log('Fetching new access token for client:', clientId, 'from token URL:', tokenUrl);
```

Comment:
Use of `:` here makes the output not flow.  Also, are spaces needed before/after each string concat point to ensure the log line looks proper?